include/swift/AST/TypeRepr.h

```diff
@@ -613,38 +613,38 @@ class NamedTypeRepr : public TypeRepr {
 
 /// \brief A protocol composite type.
 /// \code
-///   protocol<Foo, Bar>
+///   Foo & Bar
 /// \endcode
 class ProtocolCompositionTypeRepr : public TypeRepr {
   ArrayRef<IdentTypeRepr *> Protocols;
-  SourceLoc ProtocolLoc;
-  SourceRange AngleBrackets;
+  SourceLoc FirstTypeLoc;
+  SourceRange CompositionRange;
```

Comment:
Are these redundant? I suggest keeping the locations of each `&` instead. The range of the composition is just the start of the first protocol to the end of the last protocol, right?